src/frontends/pytorch/src/transforms/prim_tuple_unpack_parameter_replacer.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "prim_tuple_unpack_parameter_replacer.hpp"
+
+#include <queue>
+
+#include "openvino/frontend/pytorch/decoder.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+bool DecomposeTupleParameters::run_on_model(const std::shared_ptr<Model>& model) {
+    bool at_least_one_decomposed = false;
+    std::queue<std::shared_ptr<ov::op::v0::Parameter>> parameters;
+    for (const auto& par : model->get_parameters()) {
+        parameters.push(par);
+    }
+    while (!parameters.empty()) {
+        auto parameter = parameters.front();
+        parameters.pop();
+        auto consumers = parameter->get_output_target_inputs(0);
+        size_t num_outputs = 0;  // number of outputs in each unpack consumer should match
+        bool all_unpacks = true;
+
+        // collects all outputs per each consumer operation for this tuple Parameter
+        std::vector<OutputVector> consumer_outputs;
+
+        for (const auto& consumer : consumers) {
+            auto node = consumer.get_node()->shared_from_this();
+            auto tuple_unpack = cast_fw_node(node, "prim::TupleUnpack");
+            if (!tuple_unpack) {
+                all_unpacks = false;
+                break;
+            }
+            if (num_outputs == 0) {
+                num_outputs = node->get_output_size();
+            } else if (num_outputs != node->get_output_size()) {
+                std::cerr << "[ PT FE WARNING ] Unpack node " << node
```

Comment:
Indeed we need to inform user about this in some way. OPENVINO_DEBUG is not shown it is useless for a regular user. Can we attach a warning to a node and print it when this node is left unconverted? If not I don't know about other methods to report a problem.

Reply:
You can use `add_exception_to_fw_node` helper function to add that exception into TupleUnpack node and it will be reported as the reason why it wasn't decomposed. If you still want to leave it as console message please change "PT FE" to "PyTorch Frontend", because pt fe might confuse user.

Reply:
Done. Also handled the case when not all the consumers have prim::TupleUnpack types to attach the diagnostics to those nodes that have type prim::TupleUnpack. Tested with manual experiments, dedicated test wasn't added (don't know how to do that).